util/file_util.cc

```diff
@@ -16,7 +16,8 @@ namespace rocksdb {
 
 // Utility function to copy a file up to a specified length
 Status CopyFile(Env* env, const std::string& source,
-                const std::string& destination, uint64_t size) {
+                const std::string& destination, uint64_t size,
+                const bool sync_after_copy) {
```

Comment:
A copy is needed if the checkpoint is created in a filesystem that is different from the one containing the WAL files. For example, WAL files live in /walfiles mounted on /dev/sda, but we're creating the checkpoint in /backups mounted on /dev/sdb, which also contain the sst datafiles. When the checkpoint is created, the WAL files need to be copied from /dev/sda to /dev/sdb while the sst datafiles can be hardlinked on /dev/sdb.